Can't target Wasm web

when
```
wasm-pack build . --target web
```
throw 
```
error: Unsupported target OS! Create an issue: https://github.com/svartalf/hostname/issues/new
  --> /Users/katopz/.cargo/registry/src/github.com-1ecc6299db9ec823/hostname-0.3.1/src/lib.rs:81:9
   |
81 |         compile_error!("Unsupported target OS! Create an issue: https://github.com/svartalf/hostname/issues/new");
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0433]: failed to resolve: use of undeclared crate or module `sys`
   --> /Users/katopz/.cargo/registry/src/github.com-1ecc6299db9ec823/hostname-0.3.1/src/lib.rs:105:5
    |
105 |     sys::get()
    |     ^^^ use of undeclared crate or module `sys`
```
which
```
└── xid v1.0.2
    ├── hostname v0.3.1
```

not sure this can be fix or any workaround?

```
[INFO]: 🌀  Compiling to Wasm...
   Compiling getrandom v0.2.7
   Compiling quote v1.0.21
error: the wasm32-unknown-unknown target is not supported by default, you may need to enable the "js" feature. For more information see: https://docs.rs/getrandom/#webassembly-support
   --> /Users/katopz/.cargo/registry/src/github.com-1ecc6299db9ec823/getrandom-0.2.7/src/lib.rs:235:9
    |
235 | /         compile_error!("the wasm32-unknown-unknown target is not supported by \
236 | |                         default, you may need to enable the \"js\" feature. \
237 | |                         For more information see: \
238 | |                         https://docs.rs/getrandom/#webassembly-support");
    | |________________________________________________________________________^

error[E0433]: failed to resolve: use of undeclared crate or module `imp`
   --> /Users/katopz/.cargo/registry/src/github.com-1ecc6299db9ec823/getrandom-0.2.7/src/lib.rs:262:5
    |
262 |     imp::getrandom_inner(dest)
    |     ^^^ use of undeclared crate or module `imp`

For more information about this error, try `rustc --explain E0433`.
error: could not compile `getrandom` due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
Error: Compiling your crate to WebAssembly failed
Caused by: failed to execute `cargo build`: exited with exit status: 101
```

it's a deeper features in `rand` btw 🤔
```
    ├── rand v0.8.5
    │   ├── libc v0.2.134
    │   ├── rand_chacha v0.3.1
    │   │   ├── ppv-lite86 v0.2.16
    │   │   └── rand_core v0.6.4
    │   │       └── getrandom v0.2.7
```
to reproduce `wasm-pack build` here 👉 https://github.com/katopz/xid-rs/tree/target-wasm/examples/wasm

No worry if there's no-way to make it work(i know the pain), nothing urgent; thx for your time 🙏

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -13,11 +13,13 @@ exclude = [".github/"]
 
 [dependencies]
 crc32fast = "^1"
-hostname = "^0.3"
 md5 = "^0.7"
 once_cell = "^1"
 rand = "^0.8"
 thiserror = "^1"
+
+[target.'cfg(not(target_arch = "wasm32"))'.dependencies]
+hostname = "^0.3"
 
 [target.'cfg(target_os = "macos")'.dependencies]
 sysctl = "^0.4"

--- a/src/machine_id.rs
+++ b/src/machine_id.rs
@@ -6,9 +6,7 @@ use sysctl::{Sysctl, SysctlError};
 pub fn get() -> [u8; 3] {
     let id = match machine_id().unwrap_or_default() {
         x if !x.is_empty() => x,
-        _ => hostname::get()
-            .map(|s| s.into_string().unwrap_or_default())
-            .unwrap_or_default(),
+        _ => hostname_fallback(),
     };
 
     let mut bytes = [0_u8; 3];
@@ -19,6 +17,19 @@ pub fn get() -> [u8; 3] {
         bytes.copy_from_slice(&md5::compute(id)[0..3]);
     }
     bytes
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+fn hostname_fallback() -> String {
+    hostname::get()
+        .map(|s| s.into_string().unwrap_or_default())
+        .unwrap_or_default()
+}
+
+// `hostname` crate doesn't support wasm, so use an empty string to generate random bytes.
+#[cfg(target_arch = "wasm32")]
+fn hostname_fallback() -> String {
+    "".to_owned()
 }
 
 // https://github.com/rs/xid/blob/efa678f304ab65d6d57eedcb086798381ae22206/hostid_linux.go

